### ISSUE TITLE
IShellSettingsConfigurationProvider problem with multi-tenant in Core 2.0

### ISSUE THREAD

**Issue by crowcoder** [OrchardCMS/OrchardCore.Samples#1]
Working with Modules in Asp.Net Core 2.0 I get the error below when I enable WithTenants. The sample in this repo **does** work, but enabling tenants in Core 2.0 does not. Is there something else to do here or is multi-tenancy not yet available with 2.0?

    services.AddModules( c => c.WithTenants());

>System.InvalidOperationException: 'Cannot consume scoped service 'Orchard.Environment.Shell.IShellSettingsConfigurationProvider' from singleton 'Orchard.Environment.Shell.IShellSettingsManager'.

![image](https://user-images.githubusercontent.com/4210624/30244028-e4e1875a-9583-11e7-9465-166794a267e4.png)

(@contrivedex on twitter)

**Comment by sebastienros** [comment]
What was the issue?

**Comment by crowcoder** [comment]
I did not get it worked out, I reverted back to the NuGet package versions that were current at the time of the [youtube video](https://www.youtube.com/watch?v=LoPlECp31Oo). I think the MyGet feeds are maybe not all synced up because the OrchardCore github repo obviously works, but trying to setup a stand-alone project for just module functionality wasn't working for me because of NuGet version references.

The `c => c.WithTenants()` works in a .Net Standard 1.6 project but I got the error reported above on 2.0.

However, tenancy (in Core v1.6) isn't restricting modules (Features) by tenant. I can tell it is doing something since the proper route works, but I can still access any module from any tenant. 

Maybe I will create a minimal repo that reproduces the behavior when I get home from work.

I have enough material for my code camp presentation so no emergencies, especially since my problems are probably just my own mistakes.

Thank you for checking!
